### DISCUSSION TITLE
Which TwinCAT Runtime license is needed to run TcUnit?

### DISCUSSION THREAD

**Discussion by janbirki** [tcunit/TcUnit#215] [General]
Hi guys / @sagatowski

I just started with TcUnit and for that I set up a windows development pc with the "all-in-one" server architecture. So I have several TwinCAT builds and jenkins installed on the same pc. So far so good. I would like to buy a TwinCAT runtime license which should be located on a USB dongle (beckhoff dongle C9900-L100). My local beckhoff sales guy told me that I need to buy the license for the highest performance level alltough the unit tests are just running "local". So a license costs around 2k euros.
What kind of license are you using and if it's cheaper if you run everything in a vm? What are your experiences in this topic?

Thanks :)

**Comment by sagatowski** [comment]
Hi @janbirki! I generally only use the TC1200 license (for PLC). If you run your tests on a Beckhoff PLC, you only need to buy the license for that platform level. To make your life simpler, I recommend to just use a Beckhoff PLC as a node for running the tests.

**Comment by janbirki** [comment]
Hi @sagatowski 
Thanks for your reply. That's right. In the meantime I also considered this approach.
